api.meta.md

```diff
@@ -0,0 +1,480 @@
+```
+
+### Type: NodeId
+```
+type NodeId {
+ 	nodeId : INT
+	proofId : ProofId
+}
+```
+
+### Type: NodeTextDesc
+```
+type NodeTextDesc {
+ 	id : NodeTextId
+	result : STRING
+}
+```
+
+### Type: NodeTextId
+```
+type NodeTextId {
+ 	nodeId : NodeId
+	nodeTextId : INT
+}
+```
+
+### Type: PrintOptions
+```
+type PrintOptions {
+ 	indentation : INT
+	pure : BOOL
+	termLabels : BOOL
+	unicode : BOOL
+	width : INT
+}
+```
+
+### Type: ProblemDefinition
+```
+type ProblemDefinition {
+ 	antecTerms : List
+	functions : List
+	predicates : List
+	sorts : List
+	succTerms : List
+}
+```
+
+### Type: ProofId
+```
+type ProofId {
+ 	env : EnvironmentId
+	proofId : STRING
+}
+```
+
+### Type: ProofMacroDesc
+```
+type ProofMacroDesc {
+ 	category : STRING
+	description : STRING
+	name : STRING
+	scriptCommandName : STRING
+}
+```
+
+### Type: ProofScriptCommandDesc
+```
+type ProofScriptCommandDesc {
+ 
+}
+```
+
+### Type: Range
+```
+type Range {
+ 	end : INT
+	start : INT
+}
+```
+
+### Type: SetTraceParams
+```
+type SetTraceParams {
+ 	value : TraceValue
+}
+```
+
+### Type: ShowDocumentParams
+```
+type ShowDocumentParams {
+ 	external : BOOL
+	selection : Range
+	takeFocus : BOOL
+	uri : STRING
+}
+```
+
+### Type: ShowDocumentResult
+```
+type ShowDocumentResult {
+ 	success : BOOL
+}
+```
+
+### Type: ShowMessageParams
+```
+type ShowMessageParams {
+ 	message : STRING
+	type : MessageType
+}
+```
+
+### Type: ShowMessageRequestParams
+```
+type ShowMessageRequestParams {
+ 	actions : MessageActionItem[]
+	message : STRING
+	type : MessageType
+}
+```
+
+### Type: SortDesc
+```
+type SortDesc {
+ 	anAbstract : BOOL
+	documentation : STRING
+	extendsSorts : List
+	s : STRING
+	string : STRING
+}
+```
+
+### Type: TaskFinishedInfo
+```
+type TaskFinishedInfo {
+ 
+}
+```
+
+### Type: TaskStartedInfo
+```
+type TaskStartedInfo {
+ 
+}
+```
+
+### Type: TermActionDesc
+```
+type TermActionDesc {
+ 	category : STRING
+	commandId : TermActionId
+	description : STRING
+	displayName : STRING
+	kind : TermActionKind
+}
+```
+
+### Type: TermActionId
+```
+type TermActionId {
+ 	id : STRING
+	nodeId : NodeId
+	pio : STRING
+}
+```
+
+### Type: TermActionKind
+```
+enum TermActionKind { BuiltIn, Script, Macro, Taclet }
+```
+
+### Type: TraceValue
+```
+enum TraceValue { Off, Message, All }
+```
+
+### Type: TreeNodeDesc
+```
+type TreeNodeDesc {
+ 
+}
+```
+
+### Type: TreeNodeId
+```
+type TreeNodeId {
+ 	id : STRING
+}
+```
+
+## Endpoints
+### client/logTrace (`server ~~> client`) 
+
+```
+Client.client/logTrace( arg0 : LogTraceParams ) **async**
+```
+
+
+### client/sayHello (`server ~~> client`) 
+
+```
+Client.client/sayHello( arg0 : STRING ) **async**
+```
+
+
+### client/showDocument (`server -> client`) 
+
+```
+Client.client/showDocument( arg0 : ShowDocumentParams ) -> ShowDocumentResult
+```
+
+
+### client/sm (`server ~~> client`) 
+
+```
+Client.client/sm( arg0 : ShowMessageParams ) **async**
```

Comment:
Similarly, avoid generic argument names like `arg0`.

Reply:
Sorry, this is due to the lack of Java reflection. I think I need to compile with `-g` for debugging information. 

These documentation are generated from the Java information.

Reply:
Since Java 8, the `-parameters` switch ensures the parameter names in class files.